tests/test_logic.py

```diff
@@ -1772,7 +1772,18 @@ def test_Controller_download_new_messages_with_new_message(mocker, session, sess
     failure_signal.connect.assert_called_once_with(
         co.on_message_download_failure, type=Qt.QueuedConnection
     )
-    set_status.assert_called_once_with("Retrieving new messages", 2500)
+
+    # fake APIJobQueue's emitting main_queue_updated when the job is queued and dequeued
+    co.api_job_queue.main_queue_updated.emit(1)
+    set_status.assert_called_once_with("Retrieving 1 message", 0)
```

Comment:
Oh, I missed this. When emitting signals in tests, it is always prudent to set up a spy to wait on them. It is verbose, but again, the signals are processed asynchronously, and nothing guarantees that they'll have been received in in the interval between their emission and the assertion, especially when the assertion statement immediately follows the emission.

```py
# This is probably best placed in the "Setup/Arrange" area of the test, typically towards the beginning?
# But maybe it's better close to where it's used! You've got better context than me to make that decision!
MAX_SIGNAL_WAITING_TIME = 50
main_queue_updated_emissions = QSignalSpy(job_queue.main_queue.queue.queue_updated_signal)

# [...]

# fake APIJobQueue's emitting main_queue_updated when the job is queued and dequeued
co.api_job_queue.main_queue_updated.emit(1)
main_queue_updated_emissions.wait(MAX_SIGNAL_WAITING_TIME)
set_status.assert_called_once_with("Retrieving 1 message", 0)

Reply:
This is true for all three "Act"/"Assert" sequences, of course.

Reply:
Thanks, I missed this one as well!